src/main/java/de/srendi/advancedperipherals/common/addons/computercraft/peripheral/MEBridgePeripheral.java

```diff
@@ -828,6 +844,165 @@ public MethodResult cancelCraftingTasks(IArguments arguments) throws LuaExceptio
+                    AdvancedPeripherals.debug("[DEBUG] ==> CPU type: " + cpu.getClass().getSimpleName());
+                    
+                    // Calculate the actual completed amount using AE2 internal logic
+                    long actualCraftedAmount = totalAmount;
+                    if (cpu instanceof appeng.me.cluster.implementations.CraftingCPUCluster cpuCluster) {
+                        try {
+                            appeng.crafting.execution.CraftingCpuLogic craftingCpuLogic = cpuCluster.craftingLogic;
+                            long pending = craftingCpuLogic.getPendingOutputs(craftedItem);
+                            long active = craftingCpuLogic.getWaitingFor(craftedItem);
+                            actualCraftedAmount = totalAmount - (pending + active);
+                            
+                            AdvancedPeripherals.debug("[DEBUG] ==> AE2 Logic - Pending: " + pending + ", Active: " + active + ", Calculated Completed: " + actualCraftedAmount);
+                            
+                            // If calculated amount is 0 or negative, it means nothing is completed yet
+                            // In this case, we should force complete with the total requested amount
+                            if (actualCraftedAmount <= 0) {
+                                AdvancedPeripherals.debug("[DEBUG] ==> No items completed yet, using total amount for force completion");
+                                actualCraftedAmount = totalAmount;
+                            }
+                        } catch (Exception e) {
+                            AdvancedPeripherals.debug("[DEBUG] ==> Could not calculate precise amount: " + e.getMessage());
+                            e.printStackTrace();
+                            // Fall back to total amount if calculation fails
+                            actualCraftedAmount = totalAmount;
+                        }
+                    } else {
+                        AdvancedPeripherals.debug("[DEBUG] ==> Non-CraftingCPUCluster, using total amount: " + totalAmount);
+                    }
+                    
+                    // For force completion, we insert the full requested amount
+                    // The notification should show what we actually added to storage
+                    long amountToInsert = actualCraftedAmount;
+                    AdvancedPeripherals.debug("[DEBUG] ==> Final amount to insert and notify: " + amountToInsert);
+
+                    MEStorage storage = AEApi.getMonitor(node);
+                    long inserted = storage.insert(craftedItem, amountToInsert, Actionable.MODULATE, bridge);
+                    AdvancedPeripherals.debug("[DEBUG] ==> Actually inserted into storage: " + inserted);
+
+                    // Cancel the job without relying on AE2 notification logic
+                    cpu.cancelJob();
+                    Thread.sleep(50);
```

Comment:
Do not use `Thread.sleep` for whatever reason. It is not a reliable way to wait for something.